### PR TITLE
Fix controller reassignment order

### PR DESCRIPTION
**Pull request by xhl1988**

### CODE REVIEW

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 1]
```diff
@@ -148,9 +148,9 @@ public ControllerHelixManager(SourceKafkaClusterValidationManager srcKafkaValida
     // 2. establishing connection with server;
     // 3. getting next data snippet from server.
     _requestConfig = RequestConfig.custom()
-        .setConnectionRequestTimeout(3000)
-        .setConnectTimeout(3000)
-        .setSocketTimeout(3000)
+        .setConnectionRequestTimeout(30000)
```
Comment (fx19880617) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: why we increase the timeout here? could you add more comments here?

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 2]
```diff
@@ -707,11 +707,29 @@ public synchronized void handleLiveInstanceChange(boolean onlyCheckOffline) thro
...
             _helixAdmin.setResourceIdealState(_helixClusterName, pipeline,
                 IdealStateBuilder
                     .resetCustomIdealStateFor(_helixAdmin.getResourceIdealState(_helixClusterName, pipeline),
                         pipeline, String.valueOf(routeId), newInstanceName));
 
+            _helixAdmin.getResourceExternalView(_helixClusterName, pipeline).getStateMap(String.valueOf(routeId));
+
+            long ts1 = System.currentTimeMillis();
+            while (!isOnline(pipeline, String.valueOf(routeId))) {
+              if (System.currentTimeMillis() - ts1 > 30000) {
```
Comment (fx19880617) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: make it a parameter.

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 3]
```diff
@@ -818,6 +836,23 @@ public synchronized void handleLiveInstanceChange(boolean onlyCheckOffline) thro
     }
   }
 
+  private boolean isOnline(String topicName, String partition) {
+    LOGGER.info("isOnline for {}, {}?", topicName, partition);
```
Comment (fx19880617) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: Need better logging.

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 4]
```diff
@@ -818,6 +836,23 @@ public synchronized void handleLiveInstanceChange(boolean onlyCheckOffline) thro
     }
   }
 
+  private boolean isOnline(String topicName, String partition) {
```
Comment (fx19880617) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: better method naming? Also why it's topic/partition?

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java (hunk 1 above)
Comment (xhl1988) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: I saw increased http get request time and increased the timeout. (now we know it's due to zookeeper instability)

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java (hunk 3 above)
Comment (xhl1988) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: Changed to `isControllerOnline`

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java (hunk 4 above)
Comment (xhl1988) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: Changed to `isControllerOnline`

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java (hunk 3 above)
Comment (xhl1988) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: Updated

**Pull request merged**